src/main/java/com/byclosure/webcat/stepdefinitions/helpers/SeleniumInteractions.java

```
@@ -0,0 +1,192 @@
+                    final List<WebElement> inputs = driver.findElements(By.id(inputId));
+                    return inputs.size() > 0 ? inputs.get(0) : null;
+                }
+            }
+        }
+        return null;
+    }
+    
+    private WebElement findElementByName(String name){
+        final List<WebElement> elementsByName =  driver.findElements(By.name(name));
+        return elementsByName.size() > 0 ? elementsByName.get(0) : null;
+    }
+    
+    private WebElement findButtonByValue(String value) {
+        final List<WebElement> elements = driver.findElements(By.cssSelector("input[type='button'][value='" + value + "']"));
+        final List<WebElement> submitButtons = driver.findElements(By.cssSelector("input[type='submit'][value='" + value + "']"));
+        final List<WebElement> buttons = driver.findElements(By.cssSelector("button[value='" + value + "']"));
+        
+        elements.addAll(submitButtons);
+        elements.addAll(buttons);
+        
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findButtonByTitle(String title) {
+        final List<WebElement> elements = driver.findElements(By.cssSelector("input[type='button'][title='" + title + "']"));
+        final List<WebElement> submitButtons = driver.findElements(By.cssSelector("input[type='submit'][title='" + title + "']"));
+        final List<WebElement> buttons = driver.findElements(By.cssSelector("button[title='" + title + "']"));
+        
+        elements.addAll(submitButtons);
+        elements.addAll(buttons);
+        
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findElementById(String id) {
+        final List<WebElement> elements = driver.findElements(By.id(id));
+        return elements.size() > 0 ? elements.get(0) : null;
+    }
+    
+    private WebElement findButtonByText(String buttonText) {
+        if(null != buttonText && !"".equals(buttonText)) {
```

You can throw an exception, or you can assume that buttonText is not empty string or null, but do not support "" / null returning null - you are promoting future null pointer exception that can blow far from the place they were first created.